test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,377 @@
+        segmentStoreInstance1 = new PravegaSegmentStoreService("segmentstore1", zkUri, conURIs1.get(0));
+        assertTrue(segmentStoreInstance1.isRunning());
+        log.info("Pravega segment store instance1 details: {}", segmentStoreInstance1.getServiceDetails());
+
+        segmentStoreInstance2 = new PravegaSegmentStoreService("segmentstore2", zkUri, conURIs1.get(0));
+        assertTrue(segmentStoreInstance2.isRunning());
+        log.info("Pravega segment store instance2 details: {}", segmentStoreInstance2.getServiceDetails());
+
+        segmentStoreInstance3 = new PravegaSegmentStoreService("segmentstore3", zkUri, conURIs1.get(0));
+        assertTrue(segmentStoreInstance2.isRunning());
+        log.info("Pravega segment store instance3 details: {}", segmentStoreInstance2.getServiceDetails());
+    }
+
+    @After
+    public void tearDown() {
+        controllerInstance1.stop();
+        segmentStoreInstance1.stop();
+        controllerInstance2.stop();
+        segmentStoreInstance2.stop();
+        controllerInstance3.stop();
+        segmentStoreInstance3.stop();
+    }
+
+    @Test(timeout = 600000)
+    public void multiReaderWriterWithFailOverTest() throws Exception {
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        readWriteTest();
+
+        //scale down SSS by 1 instance
+        segmentStoreInstance1.stop();
+        Thread.sleep(60000);
+        log.info("Test with 1 SSS instance down");
+        readWriteTest();
+
+        //scale down controller by 1 instance + scale up SSS back to 2 instances
+        controllerInstance2.stop();
+        Thread.sleep(60000);
+        log.info("Test with 1 controller instance down");
+        readWriteTest();
+
+        //scale down 1 instance of both controller, SSS
+         log.info("Test with 1 controller  and 1 SSS instance down");
+         segmentStoreInstance2.stop();
+         controllerInstance3.stop();
+         Thread.sleep(60000);
+         readWriteTest();
+
+        log.info("all tests are successful");
+    }
+
+    private void readWriteTest() throws InterruptedException, ExecutionException {
+
+        String scope = "testMultiReaderWriterScope" + new Random().nextInt(Integer.MAX_VALUE);
+        String readerGroupName = "testMultiReaderWriterReaderGroup" + new Random().nextInt(Integer.MAX_VALUE);
+        //20  readers -> 20 stream segments ( to have max read parallelism)
+        ScalingPolicy scalingPolicy = ScalingPolicy.fixed(20);
+        StreamConfiguration config = StreamConfiguration.builder().scope(scope)
+                .streamName(STREAM_NAME).scalingPolicy(scalingPolicy).build();
+
+        //get Controller Uri
+        Service conService = new PravegaControllerService("controller1", null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        URI controllerUri = ctlURIs.get(0);
+        Controller controller = getController(controllerUri);
```

Comment:
looks like only controller1 is being used here, so controller2 and controller3 failure will have no effect on the client.
To try to use all controllers you need to use tcp://uri1,uri2,ur3, or pravega:// urls; see MultiControllerTest.java for how this is constructed.